source/fuzz/fuzzer_pass_apply_id_synonyms.cpp

```diff
@@ -95,14 +101,30 @@ void FuzzerPassApplyIdSynonyms::Apply() {
           // We cannot insert an extract before this instruction, so this
           // synonym is no good.
           //
-          // TODO(afd): In the case of an OpPhi, we could consider inserting the
-          //  extract instruction into the relevant parent block of the OpPhi.
-          continue;
-        }
+          // In the case of an OpPhi, we insert the extract instruction at the
+          // end of the relevant parent block.
+          if (use_inst->opcode() == SpvOpPhi) {
+            auto parentBlockId =
+                use_inst->GetSingleWordInOperand(use_in_operand_index + 1);
+            auto parentBlockInstruction =
+                GetIRContext()->get_def_use_mgr()->GetDef(parentBlockId);
+            auto parentBlock =
+                GetIRContext()->get_instr_block(parentBlockInstruction);
+            uint32_t compositeExtractId = GetFuzzerContext()->GetFreshId();
+
+            ApplyTransformation(TransformationCompositeExtract(
+                MakeInstructionDescriptor(GetIRContext(),
+                                          parentBlock->terminator()),
```

Comment:
Using the terminator is basically correct.  However, the penultimate instruction in the block might be a merge instruction (OpLoopMerge or OpSelectionMerge), in which case you'd need to insert before that instruction, since you can't have an instruction after a merge instruction but before a block terminator.  I believe BasicBlock has methods to check whether there is a merge instruction and get it if so.